pandas/tests/io/parser/test_common.py

```diff
@@ -2135,3 +2135,22 @@ def test_no_header_two_extra_columns(all_parsers):
     parser = all_parsers
     df = parser.read_csv(stream, header=None, names=column_names, index_col=False)
     tm.assert_frame_equal(df, ref)
+
+
+def test_read_csv_names_types(all_parsers):
+    # GH 34946
+    data = """\
+    1,2,3
+    4,5,6
+    7,8,9
+    10,11,12\n"""
+    parser = all_parsers
+    msg = "Names should have consistent ordering. Consider a list instead."
+    names = "QAZ"
+    with pytest.raises(ValueError, match=msg):
+        parser.read_csv(StringIO(data), names=set(names))
+
+    ref = DataFrame(data={"Q": [1, 4, 7, 10], "A": [2, 5, 8, 11], "Z": [3, 6, 9, 12]})
+    for valid_type_converter in (list, tuple):
+        df = parser.read_csv(StringIO(data), names=valid_type_converter(names))
+        tm.assert_frame_equal(df, ref)
```

Comment:
Should be a separate test.  Even so, not sure I fully understand the purpose, given that we implicitly test this in all of our other `names` tests.

Reply:
Yes that makes sense. I just included this section to be more cautious; better to get a comment about the test being too much/redundant than a not-enough-tests one.